### PR TITLE
Add openmpi v5.0.2

### PR DESCRIPTION
**Pull request by srishtih**
Fixes:

Related:

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

#### For new version streams

- [ ] The upstream project actually supports multiple concurrent versions.
- [ ] Any subpackages include the version string in their package name (e.g. `name: ${{package.name}}-compat`)
- [ ] The package (and subpackages) `provides:` logical unversioned forms of the package (e.g. `nodejs`, `nodejs-lts`)

#### For security-related PRs

- [ ] The security fix is recorded in the [advisories](https://github.com/wolfi-dev/advisories) repo

#### For version bump PRs

- [x] The `epoch` field is reset to 0

#### For PRs that add patches

- [ ] Patch source is documented

### CODE REVIEW

**Comment by srishtih** [comment]
For building Transformer Engine used in  https://github.com/chainguard-dev/image-requests/issues/787

File: openmpi.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      - hwloc-dev
+      - libevent-dev
+      - libgomp
+      - perl
+
+pipeline:
+  - uses: fetch
+    with:
+      expected-sha256: f740994485516deb63b5311af122c265179f5328a0d857a567b85db00b11e415
+      uri: https://www.open-mpi.org/software/ompi/v4.1/downloads/openmpi-${{package.version}}.tar.bz2
```
Comment (rawlingsj) [inline openmpi.yaml:29]: can we use a mangled version transform to avoid the hardcoded 4.1 in this URL?

Comment (rawlingsj) [reply openmpi.yaml:29]: i.e. similar to https://github.com/wolfi-dev/os/blob/b7fe3f67b76e0d7cdef73d1ee47a640ce0367546/gnutls.yaml#L13-L18

**Comment by mamccorm** [comment]
Hey @srishtih, looks like an old PR, do you still need this or can we close it out?

**Pull request closed without merging**